### PR TITLE
Add a Liquid filter `uri_expand`

### PR DESCRIPTION
**Pull request by knu**
This implements #853.

### CODE REVIEW

File: app/concerns/liquid_interpolatable.rb [hunk 1]
```diff
@@ -132,6 +132,42 @@ def to_uri(uri, base_uri = nil)
...
+            response = http.head(uri)
+            case response.status
+            when 301, 302, 303, 307
+              if location = response['location']
+                uri += location
+                next
+              end
+            end
+          end
+        rescue
```
Comment (cantino) [inline app/concerns/liquid_interpolatable.rb]: Seems like this could end up obscuring a bug.

**Comment by cantino** [comment]
Awesome!  Be sure to add it to the wiki too so people know about it.  Looks good to me beyond my one comment about a `rescue` that could potentially hide a bug -- what if something unexpected happens like a network error inside that block?

File: app/concerns/liquid_interpolatable.rb (hunk 1 above)
Comment (knu) [inline app/concerns/liquid_interpolatable.rb]: There are too many kinds of errors that may be raised here, but I'll try to list them as many as I can think of.

**Comment by knu** [comment]
Actually, detecting a various kinds of network/protocol errors was the top motivation to have a blanket rescue here.  A server may be down or extremely slow on their side, their DNS may have a problem, or the network where the Huginn server is located may be in trouble.  So, in such cases we don't really know what to do, so uri_expand() just returns the original URL instead of crashing.

**Comment by cantino** [comment]
Makes sense. Nice!

**Pull request merged**